### DISCUSSION TITLE
how to upload file with graphene-file-upload

### DISCUSSION THREAD

**Discussion by kevinzDzul** [graphql-python/graphene-django#1170] [Q&A] [answered]
how to save file .. is confuse for me.

**Comment by phalt** [comment]
Needs a bit more information @kevinzDzul - have you tried to do this already with graphQL mutations?

**Comment by phalt** [comment]
Check out: https://docs.graphene-python.org/en/latest/relay/mutations/#accepting-files

**Comment by kevinzDzul** [comment, marked as answer]
my mutation,
filedata[0] is array for multiple files 

    class UploadFileMtrs(graphene.Mutation):

    class Input:
        filedata = Upload()

    filedata = Upload
    nameFile = graphene.String()
    

    @staticmethod
    def mutate(self, info,  filedata=None):
        print(filedata[0])

        date_file = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")
        date = datetime.datetime.now().strftime("%Y-%m-%d_%H:%M:%S")
        dateplus = date + ".txt"
        fs = FileSystemStorage()
        fs.save(dateplus,filedata[0])

use this library in back:  https://github.com/lmcgartland/graphene-file-upload 
and use this in frond :  https://github.com/jaydenseric/apollo-upload-client

**Comment by acorrea-B** [comment]
if I a need upload a list of files, how identify the name for each one
